### PR TITLE
[Multi-sig Toolkit] Introduce user-friendly effects preview

### PR DESCRIPTION
**Pull request by manolisliolios**
## Description 

Introduces effects preview in an easy to read way in the offline signer.

Before:
Screenshot 2024-02-05 at 8 27 55 PM

After:
Screenshot 2024-02-05 at 8 28 16 PM

## Test Plan 

How did you test the new or updated feature?

---
If your changes are not user-facing and do not break anything, you can skip the following section. Otherwise, please briefly describe what has changed under the Release Notes section.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

**Review by jnaulty — APPROVED** [review]
Tested out the functionality here, and I think it's AMAZING 🤩 
We've had a lot of internal/external folks want a better way to visualize Transaction Effects, thanks @manolisliolios  for diving into this problem and surfacing with a beautiful solution 🚀

File: dapps/multisig-toolkit/src/components/preview-effects/utils.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+export const onChainAmountToFloat = (amount: string, decimals: number) => {
+	if (!decimals) return undefined;
+	const total = parseFloat(amount);
+
+	return total / Math.pow(10, decimals);
+};
+
+export const formatAddress = (address: string) => {
```
Comment (Jordan-Mysten) [inline dapps/multisig-toolkit/src/components/preview-effects/utils.ts:10]: FYI I think @mysten/sui/utils exports something similar to this. Not a big deal to replicate just figured I'd let you know!

Comment (Jordan-Mysten) [reply dapps/multisig-toolkit/src/components/preview-effects/utils.ts:10]: Oh nvm this does last 10 chars

File: dapps/multisig-toolkit/src/components/preview-effects/utils.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+export const onChainAmountToFloat = (amount: string, decimals: number) => {
+	if (!decimals) return undefined;
```
Comment (Jordan-Mysten) [inline dapps/multisig-toolkit/src/components/preview-effects/utils.ts:4]: Is it expected that `0` returns undefined? Also generally prefer null over undefined for explicit empty values

File: dapps/multisig-toolkit/src/components/preview-effects/EffectsPreview.tsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+			title: 'Raw JSON',
+		},
+	];
+
+	return (
+		<>
+			<Overview output={output} />
+			<Tabs.Root value={tab} onValueChange={setTab} className="w-full">
+				<Tabs.List className="flex overflow-x-auto border-b ">
+					{tabs.map((tab, index) => {
```
Comment (Jordan-Mysten) [inline dapps/multisig-toolkit/src/components/preview-effects/EffectsPreview.tsx:55]: Kind of odd to have the tabs defined in the list but the tab content defined explicitly below.

File: dapps/multisig-toolkit/src/components/preview-effects/PreviewCard.tsx [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+}
+
+const PreviewCard = {
+	Root,
+	Header,
+	Body,
+	Footer,
+};
+
+export default PreviewCard;
```
Comment (Jordan-Mysten) [inline dapps/multisig-toolkit/src/components/preview-effects/PreviewCard.tsx:65]: Tiny nit but I hate default exports

**Pull request merged**